### PR TITLE
Fix returned type in predict

### LINKED ISSUE #962: [Bug] Why `predict` sometimes return `(array, states)` instead of `(action, state)`? Is it a BUG?
**Important Note: We do not do technical support, nor consulting** and don't answer personal questions per email.
Please post your question on the [RL Discord](https://discord.com/invite/xhfNqQv), [Reddit](https://www.reddit.com/r/reinforcementlearning/) or [Stack Overflow](https://stackoverflow.com/) in that case.

If your issue is related to a **custom gym environment**, please use the custom gym env template.

### To Reproduce

```py
import numpy as np
import stable_baselines3 as sb
import stable_baselines3.common.base_class
from stable_baselines3.common.env_checker import check_env
import gym, gym.spaces


class MarketEnv(gym.Env):
    def __init__(self):
        super().__init__()

        self.simulate_max_steps = 100

        self.action_space = gym.spaces.Discrete(3)
        self.observation_space = gym.spaces.Box(
            low=np.array([ 0, -np.inf]),
            high=np.array([1,  np.inf]),
            dtype=np.float64
        )

    def reset(self):
        self.simulate_max_steps = 100

        return np.array([0, 0.0])

    def step(self, action):
        self.simulate_max_steps -= 1

        obs = np.array([1, 0.0])
        reward = 0.5
        done = self.simulate_max_steps == 0
        return obs, reward, done, { }

    def close(self):
        pass

    def render(self):
        pass

env = MarketEnv()
check_env(env)

model = sb.DQN('MlpPolicy', env, device='cpu')
model.learn(100)

def test_model(model: stable_baselines3.common.base_class.BaseAlgorithm, 
               env: MarketEnv, episode_nums: int) -> np.ndarray:
    for _ in range(episode_nums):
        obs = env.reset()
        done = False

        while not done:
            action, states = model.predict(obs)
            if isinstance(action, np.ndarray):
                print(type(action), action)
            obs, reward, done, info = env.step(action)

    env.reset()


test_model(model, env, 1)
```

Output:

```
<class 'numpy.ndarray'> 1
<class 'numpy.ndarray'> 2
<class 'numpy.ndarray'> 1
<class 'numpy.ndarray'> 2
<class 'numpy.ndarray'> 2
<class 'numpy.ndarray'> 1
```

### Expected behavior

`model.predict()` should return `(action, states)`, not `(np.ndarray, states)`.

### System Info

```
OS: Windows-10-10.0.19041-SP0 10.0.19041
Python: 3.9.13
Stable-Baselines3: 1.6.0
PyTorch: 1.12.0
GPU Enabled: True
Numpy: 1.21.5
Gym: 0.21.0
```

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

In some case, `predict` can return wrong type:

```python
import gym
import numpy as np
from stable_baselines3 import DQN

env = gym.make("CartPole-v1")

model = DQN("MlpPolicy", env)
model.learn(100)

obs = env.reset()
done = False

while not done:
    action, states = model.predict(obs)
    print(type(action), isinstance(action, np.ndarray))
    obs, reward, done, info = env.step(action)
```
```
<class 'numpy.int64'> False
<class 'numpy.int64'> False
<class 'numpy.int64'> False
<class 'numpy.ndarray'> True
<class 'numpy.int64'> False
<class 'numpy.int64'> False
<class 'numpy.ndarray'> True
<class 'numpy.ndarray'> True
```

closes #962

Twin PR in sb3-contrib: https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/88

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
There isn't WIP section in the changelog anymore?

File: stable_baselines3/common/policies.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def predict(
 
         # Remove batch dimension if needed
         if not vectorized_env:
-            actions = actions[0]
+            actions = actions.squeeze(0)
```
Comment (araffin) [inline stable_baselines3/common/policies.py:353]:
```suggestion
            actions = actions.squeeze(axis=0)
```

Please use explicit keyword when using numpy ;) unless you have a good reason not to?

Comment (qgallouedec) [reply stable_baselines3/common/policies.py:353]: To be fair, I wanted to ([5009949](https://github.com/DLR-RM/stable-baselines3/pull/964/commits/50099496ee3df53f0f9023ff7b700362dbe58f5b)), but I saw this

https://github.com/DLR-RM/stable-baselines3/blob/a6f5049a99a4c21a6f0bcce458ca3306cef310e0/stable_baselines3/common/distributions.py#L583

Is there a reason to you explicit keyword using numpy and not torch?

**Comment by araffin** [comment]
> There isn't WIP section in the changelog anymore?

now there is ;)

**Pull request merged**